charts/rancher-logging/rancher-logging/3.9.401+up3.9.4/templates/_helpers.tpl

```diff
@@ -75,11 +75,32 @@ true
 {{- end }}
 {{- end }}
 
-{{- define "windowsPathPrefix" }}
-{{- $temp := (default "c:\\" .Values.global.cattle.rkeWindowsPathPrefix | replace "\\" "/" | replace "//" "/") }}
-{{- if (hasSuffix "/" $temp) }}
-{{- trimSuffix "/" $temp }}
-{{- else }}
-{{- $temp }}
-{{- end }}
-{{- end }}
+{{- define "windowsPathPrefix" -}}
+{{- default "c:\\" .Values.global.cattle.rkeWindowsPathPrefix | replace "\\" "/" | replace "//" "/" -}}
+{{- end -}}
+
+{{- define "windowsInputTail" -}}
+{{- $temp := (include "windowsPathPrefix" .) -}}
+{{- if (hasSuffix "/" $temp) -}}
+{{- trimSuffix "/" $temp -}}
+{{- else -}}
+{{- $temp -}}
+{{- end -}}
+{{- end -}}
+
+{{- define "windowsKubernetesFilter" -}}
+{{- $temp := ((include "windowsPathPrefix" .) | replace ":" "" | replace "/" ".") -}}
+{{- if (hasSuffix "." $temp) -}}
+{{- $temp = trimSuffix "." $temp -}}
+{{- end -}}
+{{- printf "kubernetes.%s" $temp -}}
+{{- end -}}
+
+{{- define "windowsInputTailMount" -}}
+{{- $temp := ((include "windowsPathPrefix" .) | replace "C:" "" | replace "c:" "") -}}
+{{- if (hasSuffix "/" $temp) -}}
+{{- trimSuffix "/" $temp -}}
+{{- else -}}
+{{- $temp -}}
+{{- end -}}
+{{- end -}}
```

Comment:
**Note to reviewers:** Due to the Kubebuilder spec, these must start with `/`. This cannot use the `windowsInputTail` as a result. This has been tested on a Windows 20H2 cluster on k8s 1.20.6 with RKE.

Source: https://github.com/banzaicloud/logging-operator/blob/f18b277cbdb8b4dd4db3ac15f8dd8f50c3e28df5/pkg/sdk/api/v1beta1/node_agent_types.go#L59